File: go/leetcode/814. Binary Tree Pruning.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func pruneTree(root *TreeNode) *TreeNode {
    if root == nil{return nil}
    
    root.Right = pruneTree(root.Right)
    root.Left = pruneTree(root.Left)
    
    if root.Val != 1 && root.Left == nil && root.Right == nil {return nil}
    
    return root    
}

File: python/leetcode/814. Binary Tree Pruning.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def pruneTree(self, root: TreeNode) -> TreeNode:
        if root is None:
            return None

        root.right = self.pruneTree(root.right)
        root.left = self.pruneTree(root.left)

        if root.val != 1 and root.left is None and root.right is None:
            return None
        return root
